src/analysis/retail/hunter/shared/talents/WailingArrow.tsx

```diff
@@ -25,12 +32,37 @@ class WailingArrow extends Analyzer {
       Events.damage.by(SELECTED_PLAYER).spell(SPELLS.WAILING_ARROW_DAMAGE),
       this.onWailingDamage,
     );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.WAILING_ARROW_DAMAGE_FOCUS),
+      this.onWailingCast,
+    );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS_HUNTER.AIMED_SHOT_TALENT),
+      this.onAimedShot,
+    );
+  }
+
+  //Readiness grants Wailing Arrow the ability to reset the cooldown of Rapid Fire and 2 charges of Aimed Shot
+  onWailingCast(event: CastEvent) {
+    if (this.selectedCombatant.hasTalent(TALENTS.READINESS_TALENT)) {
+      this.spellUsable.endCooldown(SPELLS.RAPID_FIRE.id, event.timestamp);
+      this.chargesLeft = 2 - this.spellUsable.chargesAvailable(TALENTS_HUNTER.AIMED_SHOT_TALENT.id);
```

Comment:
.endCooldown has a boolean property for restoring all charges, wouldn't that be better in this case?